WebComponentGenerator fails with java.lang.IllegalStateException: Unsupported type: class com.fasterxml.jackson.databind.node.BaseJsonNode

### Description of the bug

When I add a property in a constructor of class extending the `WebComponentExporter` which takes `com.fasterxml.jackson.databind.node.ObjectNode` as default value, it is internally added with `com.fasterxml.jackson.databind.node.BaseJsonNode` type inside `com.vaadin.flow.component.WebComponentExporter#addProperty(java.lang.String, com.fasterxml.jackson.databind.node.BaseJsonNode)`. When I try to start the application I am receiving the following exception:
```
java.lang.RuntimeException: Unable to initialize com.vaadin.flow.spring.VaadinServletContextInitializer$DevModeServletContextListener
	at com.vaadin.flow.spring.VaadinServletContextInitializer$FailFastServletContextListener.contextInitialized(VaadinServletContextInitializer.java:206) ~[vaadin-spring-24.9.12.jar:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$CompositeServletContextListener.lambda$contextInitialized$0(VaadinServletContextInitializer.java:230) ~[vaadin-spring-24.9.12.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596) ~[na:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$CompositeServletContextListener.contextInitialized(VaadinServletContextInitializer.java:230) ~[vaadin-spring-24.9.12.jar:na]
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4032) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:4462) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1201) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1191) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:317) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:81) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:145) ~[na:na]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:747) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:770) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1201) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1191) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:317) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:81) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:145) ~[na:na]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:747) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:201) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:410) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:868) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.apache.catalina.startup.Tomcat.start(Tomcat.java:436) ~[tomcat-embed-core-10.1.48.jar:10.1.48]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.initialize(TomcatWebServer.java:128) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.<init>(TomcatWebServer.java:107) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getTomcatWebServer(TomcatServletWebServerFactory.java:517) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getWebServer(TomcatServletWebServerFactory.java:219) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:193) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:167) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:621) ~[spring-context-6.2.12.jar:6.2.12]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439) ~[spring-boot-3.5.7.jar:3.5.7]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318) ~[spring-boot-3.5.7.jar:3.5.7]
	at com.librade.bootstrap.gui.AltfinsGUIApplication.main(AltfinsGUIApplication.java:85) ~[classes/:na]
Caused by: java.lang.RuntimeException: Could not write exported web component module because of exception: Unsupported type: class com.fasterxml.jackson.databind.node.BaseJsonNode
	at com.vaadin.flow.server.webcomponent.WebComponentModulesWriter$DirectoryWriter.generateWebComponentsToDirectory(WebComponentModulesWriter.java:259) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.frontend.FrontendWebComponentGenerator.generateWebComponents(FrontendWebComponentGenerator.java:95) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.frontend.NodeTasks.<init>(NodeTasks.java:179) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.base.devserver.startup.DevModeInitializer.initDevModeHandler(DevModeInitializer.java:348) ~[vaadin-dev-server-24.9.12.jar:na]
	at com.vaadin.base.devserver.DevModeHandlerManagerImpl.initDevModeHandler(DevModeHandlerManagerImpl.java:104) ~[vaadin-dev-server-24.9.12.jar:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$DevModeServletContextListener.failFastContextInitialized(VaadinServletContextInitializer.java:593) ~[vaadin-spring-24.9.12.jar:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$FailFastServletContextListener.contextInitialized(VaadinServletContextInitializer.java:202) ~[vaadin-spring-24.9.12.jar:na]
	... 41 common frames omitted
Caused by: java.lang.IllegalStateException: Unsupported type: class com.fasterxml.jackson.databind.node.BaseJsonNode
	at com.vaadin.flow.server.webcomponent.WebComponentGenerator.getJSTypeName(WebComponentGenerator.java:317) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.webcomponent.WebComponentGenerator.getAttributeChange(WebComponentGenerator.java:229) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.webcomponent.WebComponentGenerator.getReplacementsMap(WebComponentGenerator.java:166) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.webcomponent.WebComponentGenerator.generateModule(WebComponentGenerator.java:137) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.webcomponent.WebComponentGenerator.generateModule(WebComponentGenerator.java:103) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.webcomponent.WebComponentModulesWriter.generateModule(WebComponentModulesWriter.java:140) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.webcomponent.WebComponentModulesWriter.writeWebComponentToDirectory(WebComponentModulesWriter.java:127) ~[flow-server-24.9.12.jar:24.9.12]
	at com.vaadin.flow.server.webcomponent.WebComponentModulesWriter.lambda$writeWebComponentsToDirectory$0(WebComponentModulesWriter.java:98) ~[flow-server-24.9.12.jar:24.9.12]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[na:na]
	at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1715) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[na:na]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[na:na]
	at com.vaadin.flow.server.webcomponent.WebComponentModulesWriter.writeWebComponentsToDirectory(WebComponentModulesWriter.java:100) ~[flow-server-24.9.12.jar:24.9.12]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
	at com.vaadin.flow.server.webcomponent.WebComponentModulesWriter$DirectoryWriter.generateWebComponentsToDirectory(WebComponentModulesWriter.java:246) ~[flow-server-24.9.12.jar:24.9.12]
	... 47 common frames omitted
```

This happens because `com.vaadin.flow.server.webcomponent.WebComponentGenerator#getJSTypeName` doesn't expect the `BaseJsonNode` but `com.fasterxml.jackson.databind.node.ObjectNode`.

This issue is related to issue [#22935](https://github.com/vaadin/flow/issues/22935) where I reported the similar problem which happened at earlier stage. I only got a chance to retest it now and encountered the above mentioned issue. 

### Expected behavior

Application starts without exception.

### Minimal reproducible example

1. Create an embeddable flow component by extending WebComponentExporter.
2. Add a property with jackson BaseJsonNode default value, e.g. `addProperty("tabData", JsonNodeFactory.instance.objectNode())`
3. Try to start application.

### Versions

- Vaadin / Flow version: 24.9.11

--- a/flow-server/src/main/java/com/vaadin/flow/server/webcomponent/WebComponentGenerator.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/webcomponent/WebComponentGenerator.java
@@ -26,6 +26,7 @@ import java.util.Set;
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.BaseJsonNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import org.apache.commons.io.IOUtils;
 
@@ -310,7 +311,8 @@ public class WebComponentGenerator {
                 || ArrayNode.class.isAssignableFrom(propertyData.getType())) {
             return "Array";
         } else if (JsonValue.class.isAssignableFrom(propertyData.getType())
-                || ObjectNode.class.isAssignableFrom(propertyData.getType())) {
+                || BaseJsonNode.class
+                        .isAssignableFrom(propertyData.getType())) {
             return "Object";
         } else {
             throw new IllegalStateException(

--- a/flow-server/src/test/java/com/vaadin/flow/server/webcomponent/WebComponentGeneratorTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/webcomponent/WebComponentGeneratorTest.java
@@ -18,6 +18,7 @@ package com.vaadin.flow.server.webcomponent;
 
 import java.util.Map;
 
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
 import org.hamcrest.MatcherAssert;
 import org.junit.Assert;
 import org.junit.Test;
@@ -215,5 +216,54 @@ public class WebComponentGeneratorTest {
                 MyComponent component) {
 
         }
+    }
+
+    public static class JacksonPropertyExporter
+            extends WebComponentExporter<MyComponent> {
+
+        public JacksonPropertyExporter() {
+            super("jackson-tag");
+            addProperty("jsonData", JsonNodeFactory.instance.objectNode());
+        }
+
+        @Override
+        public void configureInstance(WebComponent<MyComponent> webComponent,
+                MyComponent component) {
+
+        }
+    }
+
+    @Test
+    public void jacksonBaseJsonNodeProperty_generatesCorrectReplacements() {
+        JacksonPropertyExporter exporter = new JacksonPropertyExporter();
+
+        Map<String, String> replacementsMap = WebComponentGenerator
+                .getReplacementsMap("jackson-tag",
+                        new WebComponentExporter.WebComponentConfigurationFactory()
+                                .create(exporter).getPropertyDataSet(),
+                        "/foo", false, null);
+
+        String attributeChange = replacementsMap.get("AttributeChange");
+        MatcherAssert.assertThat(attributeChange,
+                containsString("_deserializeValue(value, Object)"));
+
+        String propertyDefaults = replacementsMap.get("PropertyDefaults");
+        MatcherAssert.assertThat(propertyDefaults, containsString("{}"));
+
+        String propertyMethods = replacementsMap.get("PropertyMethods");
+        MatcherAssert.assertThat(propertyMethods,
+                containsString("get ['jsonData']()"));
+        MatcherAssert.assertThat(propertyMethods,
+                containsString("set ['jsonData'](value)"));
+    }
+
+    @Test
+    public void jacksonBaseJsonNodeProperty_generateModuleSucceeds() {
+        String module = WebComponentGenerator
+                .generateModule(new DefaultWebComponentExporterFactory<>(
+                        JacksonPropertyExporter.class), "", null);
+        Assert.assertNotNull(module);
+        MatcherAssert.assertThat(module,
+                containsString("customElements.define('jackson-tag',"));
     }
 }

